xds/src/main/java/io/grpc/xds/WeightedTargetLoadBalancer.java

```diff
@@ -191,17 +199,17 @@ private ChildHelper(String name) {
     @Override
     public void updateBalancingState(final ConnectivityState newState,
         final SubchannelPicker newPicker) {
-      syncContext.execute(new Runnable() {
-        @Override
-        public void run() {
-          if (!childBalancers.containsKey(name)) {
-            return;
-          }
-          currentState = newState;
-          currentPicker = newPicker;
-          updateOverallBalancingState();
-        }
-      });
+      if (!childBalancers.containsKey(name)) {
```

Comment:
`childBalancers` is mutated within handleResolvedAddressesInternal(). The code looks like it functions correctly, but could be a source of bug in the future. It looks like this condition is mostly just an optimization to avoid `updateOverallBalancingState()`. Would it make sense to unconditionally update `currentState`/`currentPicker` and move this condition later in the method?

Reply:
Like so?